lib/autobuild/import/archive.rb

```diff
@@ -259,10 +259,33 @@ def download_from_url(package)
         # @return [Boolean] true if a new file was downloaded, false otherwise
         def update_cache(package)
             updated = download_from_url(package)
-            @cachefile_digest = Digest::SHA1.hexdigest File.read(cachefile)
+            @cachefile_digest = read_cachefile_digest
+
+            if @expected_digest and @expected_digest != @cachefile_digest
+                raise ConfigException, "The archive #{@url.to_s} does not match the digest provided"
+            end
+
             updated
         end
 
+        def read_cachefile_digest
+            Digest::SHA1.hexdigest File.read(cachefile)
+        end
+
+        def digest
```

Comment:
Is there really a point in having `digest` if you standardize the whole fingerprint idea as `fingerprint` ?

Reply:
The idea I had was that fingerprint was a general property of any importer, while each specific importer type has it's own hash methods for different purposes, the digest method was used to expose the inner cached digest and use it in the versions file. Now that I understand better the idea behind versions the digest probably shouldn't be in it either. I'll remove it.